src/main/java/com/snowflake/kafka/connector/records/RecordService.java

```diff
@@ -301,6 +322,12 @@ void putKey(SinkRecord record, ObjectNode meta) {
       return;
     }
 
+    if (record.keySchema() == null) {
```

Comment:
why do we need this?

Reply:
I am concerned about this change.

Reply:
discussed in this comment - ensuring we throw a snowflake error instead of NPE. We should never get into this situation, but i want to add this just in case

Reply:
keyschemas are often null
I dont think we should do this. this seems like a behavior change.

Reply:
keyschemas are often null, we update it in handleNativeRecords earlier in the call stack. if the key schema is null in this method the next if statement throws an NPE. 

The only behavior change here is to change the exception thrown from an NPE to a Snowflake Error

Reply:
I See, thanks. Sounds like a harmless change but its recommended to not have refactoring/fixes in the same PR as features. 

This is a small enough change to go through but it just raises more questions from reviewers POV

Reply:
I see, it was caught during testing that method since i wanted to give all possible null scenarios